ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp

```diff
@@ -30,19 +31,18 @@ using OutputWidth = uint32_t;
 using Result = std::tuple<ttnn::Tensor, OutputHeight, OutputWidth, ttnn::Tensor, std::optional<ttnn::Tensor>>;
 
 struct Conv2dConfig {
-    MathFidelity math_fidelity = MathFidelity::HiFi4;
     DataType dtype = DataType::BFLOAT16;
     DataType weights_dtype = DataType::BFLOAT16;
-    bool math_approx_mode_enabled = true;
-    bool fp32_dest_acc_enabled = false;
-    bool packer_l1_accum_enabled = false;
     string activation = "";
     uint32_t input_channels_alignment = 32;
     bool deallocate_activation = false;
     bool reallocate_halo_output = false;
-    uint32_t act_block_h_override = 0; // This argument is ignored when shard_layout == WIDTH_SHARDED.
-    uint32_t act_block_w_div = 1; //Amount by which the maximum possible act_block_width is divided. Max act_block_w = (in_channels * window_w * window_h)/total_num_cores;
-                                  //Ignored when shard_layout == HEIGHT_SHARDED or BLOCK_SHARDED
+
+    //Amount by which the maximum possible act_block is divided along a certain dimension.
+    //Divided along height for Height & Block Sharding. Divided along width for Width Sharding.
+    //When not set, chooses the smallest possible act block size(lowest perf). Setting to 1 chooses largest possible (highest perf)
+    std::optional<uint32_t> act_block_div = 1;
```

Comment:
1 still means use 1 as a value. It's just that 1 uses the largest possible act block size. null opt is the exception. Another option would be use -1 instead of nulllopt for using the smallest possible act block size.

Reply:
isn't for height sharding 1 the smallest possible value?

Reply:
ok I see thx.